### PR TITLE
Fix test running in docker (close #5709)

### LINKED ISSUE #5709: Testruns using docker container for the latest (1.10.0-rc.1) candidate give babel runtime errors
### What is your Test Scenario?

 Run tests in docker container
### What is the Current behavior?

 ```
$ docker-compose -f docker-compose.yml -p buildkite806bfe3f2e8b4a85ace8c7b98b5436dc run --name buildkite806bfe3f2e8b4a85ace8c7b98b5436dc_testcafe_build_11 --rm testcafe
--
  | The "src" and "browsers" options from the configuration file will be ignored.
  | ERROR Cannot prepare tests due to an error.
  |  
  | Error: [BABEL] /tests/tests/td_tests/rates_terms_page/rates-terms-page-FL-tests.spec.js: Failed to resolve "@babel/runtime" relative to "/tests" (While processing: "base$0")

```
### What is the Expected behavior?

 Runs the tests 
### What is your web application and your TestCafe test code?

 
<details>
<summary>Your website URL (or attach your complete example):</summary>

</details>

<details>
<summary>Your complete test code (or attach your test files):</summary>

 
```js
 
```
</details>

<details>
<summary>Your complete configuration file (if any):</summary>

```

```
</details>

<details>
<summary>Your complete test report:</summary>

```

```
</details>

<details>
<summary>Screenshots:</summary>

```

```
</details>

### Steps to Reproduce:

1. Go to my website ...
3. Execute this command...
4. See the error...
 
### Your Environment details:
 
* testcafe version:                   
* node.js version:                    
* command-line arguments:  
* browser name and version: 
* platform and version:          
* other:

### PR DESCRIPTION
**Pull request by miherlosev**
Changes:
* fix problem with path resolving in Babel 7 (change compilation settings).
* add TestCafe test run in the created docker image using the command-line interface (`gulp/docker/run-functional-test-via-command-line.js`).
* move docker related stuff from `Gulpfile.js` file to the `gulp/docker` folder.

### CODE REVIEW

**Review by AlexKamaev — APPROVED** [review]

File: docker/testcafe-docker.sh [hunk 1]
```diff
@@ -6,4 +6,5 @@ dbus-daemon --session --fork
 Xvfb :1 -screen 0 "${XVFB_SCREEN_WIDTH}x${XVFB_SCREEN_HEIGHT}x24" >/dev/null 2>&1 &
 export DISPLAY=:1.0
 fluxbox >/dev/null 2>&1 &
-node /usr/lib/node_modules/testcafe/bin/testcafe-with-v8-flag-filter.js --ports 1337,1338 "$@"
+cd /usr/lib/node_modules/testcafe
+node bin/testcafe-with-v8-flag-filter.js --ports 1337,1338 "$@"
```
Comment (AndreyBelym) [inline docker/testcafe-docker.sh:10]: We cannot change the current working directory here. CI subsystems set the current working directory to the project root before starting TestCafe, and users may expect that the project root is the current directory when using relative file names in test files.

**Review by AndreyBelym — CHANGES_REQUESTED** [review]

File: gulp/docker/run-functional-test-via-command-line.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    const tmpDir   = tmp.dirSync();
+    const testsDir = path.join(__dirname, '../../test/docker/testcafe-fixtures');
+
+    return copy(testsDir, tmpDir.name).then(() => {
+        const resultTmpDirPath = osFamily.win ? correctPathOnWindows(tmpDir.name) : tmpDir.name;
+
+        if (osFamily.linux)
+            setAccessPrivileges(resultTmpDirPath);
+
+        const cmd = `docker run -v ${resultTmpDirPath}:/tests -i ${publishRepository}:${packageInfo.version} chromium /tests/*.js`;
```
Comment (wentwrong) [inline gulp/docker/run-functional-test-via-command-line.js:43]: We can safely remove the container after its usage is done (via [--rm](https://docs.docker.com/engine/reference/run/#clean-up---rm)) and replace the `chromium` browser alias with `all`.

**Review by wentwrong — COMMENTED** [review]
After execution of the `docker-test` gulp task there are some used docker images and containers, which I think should be cleaned up.

File: gulp/docker/run-functional-test-via-command-line.js (hunk 2 above)
Comment (miherlosev) [reply gulp/docker/run-functional-test-via-command-line.js:43]: This test covers test running on docker via command-line interface. It's necessary to use all installed browsers.
However, removing the containers on task end is a good idea.

**Review by AndreyBelym — APPROVED** [review]

**Pull request merged**